src/astro/sql/operators/transform.py

```diff
@@ -0,0 +1,197 @@
+import inspect
+from typing import Callable, Dict, Iterable, Mapping, Optional, Union
+
+from airflow.decorators.base import DecoratedOperator, task_decorator_factory
+from sqlalchemy.sql.functions import Function
+
+from astro.databases import create_database
+from astro.sql.table import Table
+from astro.utils.dataframe_function_handler import DataframeFunctionHandler
+from astro.utils.sql_handler import SQLHandler
+from astro.utils.table_handler import TableHandler
+
+
+class TransformOperator(
+    SQLHandler, DataframeFunctionHandler, DecoratedOperator, TableHandler
+):
+    """Handles all decorator classes that can return a SQL function"""
+
+    # todo: Add docstrings
+    def __init__(
+        self,
+        conn_id: Optional[str] = None,
+        autocommit: bool = False,
+        parameters: Optional[dict] = None,
+        handler: Optional[Function] = None,
+        database: Optional[str] = None,
+        schema: Optional[str] = None,
+        warehouse: Optional[str] = None,
+        role: Optional[str] = None,
+        raw_sql=False,
+        sql="",
+        **kwargs,
+    ):
+        self.kwargs = kwargs or {}
+        self.op_kwargs: Dict = self.kwargs.get("op_kwargs") or {}
+        self.output_table: Optional[Table] = self.op_kwargs.pop("output_table", None)
+        self.handler = self.op_kwargs.pop("handler", None)
+        self.sql = sql
+
+        super().__init__(
+            sql=sql,
+            autocommit=autocommit,
+            handler=handler,
+            parameters=parameters,
+            conn_id=self.op_kwargs.pop("conn_id", conn_id),
+            database=self.op_kwargs.pop("database", database),
+            schema=self.op_kwargs.pop("schema", schema),
+            warehouse=self.op_kwargs.pop("warehouse", warehouse),
+            role=self.op_kwargs.pop("role", role),
+            raw_sql=raw_sql,
+            **kwargs,
+        )
+
+    def execute(self, context: Dict):
+        self._set_variables_from_first_table()
+        self.database_impl = create_database(self.conn_id)
+
+        # Find and load dataframes from op_arg and op_kwarg into Table
+        self.create_output_table(self.output_table_name)
+        self.load_op_arg_dataframes_into_sql()
+        self.load_op_kwarg_dataframes_into_sql()
+
+        # Get SQL from function and render templates in the SQL String
+        self.read_sql_from_function()
+        self.move_function_params_into_sql_params(context)
+        self.template(context)
+        self.parameters = self.database_impl.process_sql_parameters(self.parameters)  # type: ignore
+        self.handle_schema()
+
+        if self.raw_sql:
```

Comment:
What is/are the case where we will be using raw_sql with transform op?

Reply:
This feels a little off, decorators shouldn't be returning SQL query results, right?